File: Front-end/app/lib/models/Facility.dart
class Facility {
  final String? uuid;
  final String? name;
  final String? address;

  Facility({
    this.uuid,
    this.name,
    this.address,
  });

  factory Facility.fromJson(Map<String, dynamic> json) {
    return Facility(
      uuid: json['uuid'],
      name: json['name'],
      address: json['address'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'uuid': uuid,
      'name': name,
      'address': address,
    };
  }
}

File: Back-end/gym-server/src/main/java/com/lht/pojo/Facility.java
package com.lht.pojo;

import org.hibernate.annotations.Generated;
import org.hibernate.annotations.GenerationTime;
import jakarta.persistence.*;
import lombok.*;

import java.util.UUID;

@Entity
@Table(name = "facility")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Facility {

    @Id
    @Column(columnDefinition = "uuid", updatable = false, nullable = false)
    @Generated(GenerationTime.INSERT)
    private UUID uuid;

    @Column(unique = true, nullable = false, length = 100)
    private String name;

    @Column(nullable = false, length = 255)
    private String address;
}
